### PR TITLE
clarification about physical qubit declaration

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes https://github.com/Qiskit/openqasm/issues/198

### CODE REVIEW

File: source/language/types.rst [hunk 1]
```diff
@@ -69,6 +69,13 @@ significantly since there is no need for quantum memory management.
 However, it also means that users or compiler have to explicitly manage
 the quantum memory.
 
+.. code-block:: c
+
+   // Declare a qubit
+   qubit gamma;
+   // Declare a qubit with a Unicode name
+   qubit γ;
+
```
Comment (zachschoenfeld33) [inline source/language/types.rst:78]: Pls add the qubit register line: `qubit[20] qubit_array;` Seems you've deleted this

File: source/language/types.rst [hunk 2]
```diff
@@ -69,6 +69,13 @@ significantly since there is no need for quantum memory management.
 However, it also means that users or compiler have to explicitly manage
 the quantum memory.
 
+.. code-block:: c
+
+   // Declare a qubit
+   qubit gamma;
+   // Declare a qubit with a Unicode name
+   qubit γ;
```
Comment (1ucian0) [inline source/language/types.rst:77]:
```suggestion
   qubit γ;
   // Declare a qubit register with 20 qubits
   qubit[20] qubit_array;
```

File: source/language/types.rst (hunk 1 above)
Comment (1ucian0) [reply source/language/types.rst:80]: added!

File: source/language/types.rst [hunk 3]
```diff
@@ -78,16 +87,16 @@ references given by ``$0``, ``$1``, ..., ``$n-1``. These qubit types are
 used in lower parts of the compilation stack when emitting physical
 circuits.
 
+The language OpenQASM3 does not impose any limitation on how the discrete space of physical qubits is organized. In other words, it is possible to declared any loose physical qubit:
```
Comment (zachschoenfeld33) [inline source/language/types.rst:90]:
```suggestion
The language OpenQASM3 does not impose any limitation on how the discrete space of physical qubits is organized. In other words, it is possible to declare any loose physical qubit:
```

Comment (zachschoenfeld33) [inline source/language/types.rst:90]:
```suggestion
There is no limitation on how the discrete space of physical qubits is organized. In other words, it is possible to declare any loose physical qubit:
```

**Comment by zachschoenfeld33** [comment]
Once more minor change @1ucian0. Then update w `master` and should be good to go.

File: source/language/types.rst [hunk 4]
```diff
@@ -78,16 +87,16 @@ references given by ``$0``, ``$1``, ..., ``$n-1``. These qubit types are
...
-   qubit gamma;
-   // Declare a qubit with a Unicode name
-   qubit γ;
-   // Declare a qubit register with 20 qubits
-   qubit[20] qubit_array;
-   // CNOT gate between physical qubits 0 and 1
-   CX $0, $1;
+   // Declare two physical qubits
+   qubit $3;
+   qubit $12;
```
Comment (blakejohnson) [inline source/language/types.rst:96]: Do physical qubits need to be declared? Seems like I should be able to directly use physical qubit indices without declaring them first...

File: source/language/types.rst [hunk 5]
```diff
@@ -69,6 +69,15 @@ significantly since there is no need for quantum memory management.
...
 the quantum memory.
 
+.. code-block:: c
+
+   // Declare a qubit
+   qubit gamma;
+   // Declare a qubit with a Unicode name
+   qubit γ;
+   // Declare a qubit register with 20 qubits
+   qubit[20] qubit_array;
```
Comment (stevenheidel) [inline source/language/types.rst:79]: @blakejohnson Are we allowing this syntax for qubit registers or do we prefer `array[qubit, 20]`?

Comment (blakejohnson) [reply source/language/types.rst:79]: Both are allowed. The types WG will need to discuss whether they are equivalent.

**Comment by blakejohnson** [comment]
@ajavadia can you weigh in here?

**Comment by blakejohnson** [comment]
I believe the decision is that physical qubits are not declared. We can re-open this PR is we decide to revisit that decision.

**Pull request closed without merging**